google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -241,6 +254,7 @@ protected ServiceOptions(Class<? extends ServiceFactory<ServiceT, OptionsT>> ser
         getFromServiceLoader(rpcFactoryClass, getDefaultRpcFactory()));
     serviceRpcFactoryClassName = serviceRpcFactory.getClass().getName();
     clock = firstNonNull(builder.clock, Clock.defaultClock());
+    transportOptions = firstNonNull(builder.transportOptions, getDefaultTransportOptions());
```

Comment:
Oh you're totally right - I knew this but somehow overlooked it. It seems worth it to change now.